src/proto/streams/stream.rs

```diff
@@ -212,6 +212,12 @@ impl Stream {
 
     /// Returns true if the stream is closed
     pub fn is_closed(&self) -> bool {
+        // If the stream has been reset by the peer then it will
+        // still have frames to send, but should count as closed
+        if self.state.is_non_local_reset() {
```

Comment:
Hm, I kind of wonder if this is bandaiding over another issue, which is that since the remote has reset this stream, then we should have cleared `pending_send` and `buffered_send_data`, right?

Reply:
Yeah, that makes sense. Just wasn't sure if that was the right thing to do

Reply:
@seanmonstar - I pushed the change for `clearing pending_send` + `buffered_send_data` to a branch: https://github.com/DarrenTsung/h2/commit/6ee879630febb1b0a186054fbdc169e5db251e99 because I'm having some issues with `pop_frame()` getting called with streams that are reset and therefore don't have any frames to pop. The only exception is apparently scheduled reset streams, hitting this `expect("must be scheduled to reset")` (https://github.com/DarrenTsung/h2/blob/clear-send-buffer-on-remote-reset/src/proto/streams/prioritize.rs#L659).

Do you know why the connection is expecting the stream closed by remote reset to pop_frame?

Reply:
The `pending_send` buffer is always used in conjunction with the `Prioritize.pending_send` queue. So, if there are items in `pending_send`, it is also true the this stream is in `pending_send` queue. Then, in `poll_complete`, when the stream is taken from the queue, it's a logic error that the stream was queued but didn't have anything to write.

Reply:
Ah okay. If this stream which is already in the `pending_send` buffer & queue receives a reset, then we should clear buffer + remove from queue.

Weird thing is that the `clear_queue()` method called by `recv_err()` only clears the buffer + doesn't remove it from the queue. How did streams that receive an error not hit this case where the buffer was cleared, but they were still queued? (Maybe they weren't queued + didn't have anything in the buffer?).

Anyways, I can see two solutions to this: 
1. Remove from queue in the `clear_queue()` (which already clears the buffer). This would require some sort of `remove()` method implemented for `store::Queue`. We would need to traverse the entire queue to find the matching key for the stream, which is not ideal.
2. Check if the stream `is_reset()` when popping a stream from the queue. If it is, we can just `continue;` and move on to the next stream.

Any preferences? I'll move on with implementing 2 first and seeing if it resolves the problem, should be a three-line change.

Reply:
I would say option 2 is best. This is a rare(ish) case, so it does not seem worth to require a linear traversal or change the queue to a doubly linked list.